packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2496,6 +2496,16 @@ export class ContainerRuntime
 		return this.dataStores.createDetachedDataStoreCore(pkg, false);
 	}
 
+	public getDataStore(dataStoreChannel: IFluidDataStoreChannel): IDataStore {
```

Comment:
Exposing this on container runtime is not a good idea.
- First of all, `IFluidDataStoreChannel` is an internal interface and we removed its exposure when we moved to the new pattern of using `IDataStore`.
- The whole point of creating `IDataStore` and not exposing `trySetAlias` directly on the internal runtime object was to allow aliasing only to the client that created it. As you pointed out in the description, this means a data store can be aliased anytime by any client.
This increases the surface area of aliasing collision. Currently, a collision can happen when 2+ clients create a new data store and try to alias them to the same name. The collision handling is straightforward - the losing client can either choose a different alias or get the already aliased data store (and throw away its created data store). Also, since aliasing happens after creation, throwing away a data store is not expensive because it would have 0 to very less data.
With this change, the collision handling changes. Since a data store can be aliased any time, the losing client cannot simply throw away the data store. It also cannot assume that the client that won, aliased the same data store. It becomes really complicated.

@andre4i to correct me if I am wrong and add anything that I missed.

Reply:
Yeah, Navin nailed the second part. There are a few safeguards in terms of aliasing idempotency and such, but they all assume they're going to happen on the same object. Allowing the conversion of a channel to an IDataStore is not necessarily dangerous, but it just complicates things and likely makes things more confusing for customers (regular aliasing already kind of is).

Reply:
Closing as this is not the direction that we want to head in.